### PR TITLE
fix extents array constructor in constant expressions

### LINKED ISSUE #277: Extents array-based constructor not usable in constant expressions for C++14/17
https://godbolt.org/z/GKPj7dKc8

### PR DESCRIPTION
**Pull request by oliverlee**
fixes #277

### CODE REVIEW

**Comment by mhoemmen** [comment]
Thanks for the contribution!  May I ask how and with what compilers you tested this?

**Comment by oliverlee** [comment]
This is what CMake picked up:
 /usr/bin/clang++ --version                                                           Ubuntu clang version 17.0.0 (++20230529052613+cd21c0d30c08-1~exp1~20230529172756.652)

File: include/experimental/__p0009_bits/extents.hpp [hunk 1]
```diff
@@ -167,7 +167,7 @@ template <> struct index_sequence_scan_impl<0> {
 // all static values.
 
 template <class T, size_t N> struct possibly_empty_array {
-  T vals[N];
+  T vals[N]{};
```
Comment (mhoemmen) [inline include/experimental/__p0009_bits/extents.hpp:170]: @crtrott I think this is harmless and might actually be _more_ conforming (it certainly looks more like the exposition-only `array` in the `extents` wording), but please do take a look; thanks!

File: compilation_tests/ctest_extents_ctors.cpp [hunk 2]
```diff
@@ -293,3 +293,7 @@ MDSPAN_STATIC_TEST(
 MDSPAN_STATIC_TEST(
   Kokkos::extents<size_t, Kokkos::dynamic_extent, 4>{ 7, 4 }.extent( 0 ) == 7
 );
+
+MDSPAN_STATIC_TEST(
+  Kokkos::dextents<size_t, 1>{std::array<std::size_t, 1>{1}}.extent(0) == 1
```
Comment (mhoemmen) [inline compilation_tests/ctest_extents_ctors.cpp:298]: Thanks for adding a unit test! : - )

**Review by mhoemmen — APPROVED** [review]

**Comment by mhoemmen** [comment]
Thanks @dalg24 ! : - )

**Pull request merged**